### PR TITLE
Add a level_id param to level_view_options, to make them level-specific

### PR DESCRIPTION
**Pull request by balderdash**
`level_view_options` are set in a bunch of different controllers, and it would make things messier if I tried to move all their logic into levels_helper. Instead, I'm allowing controllers to set them by level, so that `script_levels_controller` (and anything else that needs to handle multiple levels) will be able to set options for each of the levels being included in appOptions.

### CODE REVIEW

File: dashboard/app/controllers/level_sources_controller.rb [hunk 1]
```diff
@@ -22,15 +23,15 @@ def show
       # embed the play area and code area for viewing solutions when contributing hints
     else
       # sharing
-      level_view_options hide_source: true
+      level_view_options @level_source.level_id, hide_source: true
```
Comment (Hamms) [inline dashboard/app/controllers/level_sources_controller.rb]: I vote we take this opportunity to do a better job of [enforcing our styleguide](https://github.com/bbatsov/ruby-style-guide#no-dsl-parens) re: parens.

File: dashboard/test/controllers/level_sources_controller_test.rb [hunk 2]
```diff
@@ -35,7 +35,7 @@ class LevelSourcesControllerTest < ActionController::TestCase
     assert_equal true, app_options[:white_background]
     assert_equal [], app_options[:callouts]
 
-    options = assigns(:level_view_options)
+    options = assigns(:level_view_options_map)[@level_source.level_id]
```
Comment (Hamms) [inline dashboard/test/controllers/level_sources_controller_test.rb:38]: why this change?

**Comment by Hamms** [comment]
this LGTM after comments!

File: dashboard/test/controllers/level_sources_controller_test.rb (hunk 2 above)
Comment (balderdash) [inline dashboard/test/controllers/level_sources_controller_test.rb:38]: While moving level view options into view_options_helper.rb, I changed the instance variable from `@level_view_options` to `@level_view_options_map`, which can contain a LevelViewOptions struct for each level.
